webapp/src/main/java/com/plhal/ares/webapp/AresApiProperties.java

```diff
@@ -0,0 +1,14 @@
+package com.plhal.ares.webapp;
+
+import com.plhal.ares.model.DataRepositoryProperties;
+
+
+public class AresApiProperties {
+
+    private final DataRepositoryProperties DATA_REPOSITORY_PROPERTIES;
```

Comment:
pozor, toto neni konstanta. jde o klasickou promennou.

Reply:
Dival jsem se na to a uplne nerozumim, proc to konstanta byt nemuze? Tuto promenou inicializuji v konstruktoru a nikde jinde do teto promene nemohu priradit jiny objekt. To podle me splnuje pozadavky na konstantu? Dekuji

Reply:
Protoze to konstanta neni. Konstanta se v jave definuje jako "private" static final (fyzicky ji nejde priradit jina hodnota a patri tride ne instanci). Pokud byste to tak definoval, tak by to bylo v poradku. Nicmene tech konfiguraci muze byt vic. Predstavte si, ze budete chrit v ramci projektu nahodit vice aplikaci nebo vice instanci data repozitory a mit v nich ruzne url v ruznych properties. V neposledni rade ma java takovou peknou vec (jmx - pripadne se podivejte). Pokud byste zvrejnil tuto beanu pres jmx, tak ji muzete za behu aplikace menit hodnoty (proto ani neni treba definovt tuto vlastnost jako final).
Pekne je to shnuto napriklad [zde](https://www.baeldung.com/java-constants-good-practices) nebo v oficialni dokumentaci k jave.